tests/oauth2/test_service_account.py

```diff
@@ -0,0 +1,199 @@
+            SERVICE_ACCOUNT_JSON_FILE)
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._service_account_email == info['client_email']
+        assert credentials._token_uri == info['token_uri']
+
+    def test_from_service_account_file_args(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+
+        credentials = service_account.Credentials.from_service_account_file(
+            SERVICE_ACCOUNT_JSON_FILE, subject='subject',
+            scopes=['email', 'profile'], additional_claims={'meta': 'data'})
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._service_account_email == info['client_email']
+        assert credentials._token_uri == info['token_uri']
+        assert credentials._scopes == ['email', 'profile']
+        assert credentials._subject == 'subject'
+        assert credentials._additional_claims['meta'] == 'data'
+
+    def test_default_state(self):
+        assert not self.credentials.valid
+        # Expiration hasn't been set yet
+        assert not self.credentials.expired
+        # Scopes haven't been specified yet
+        assert self.credentials.requires_scopes
+
+    def test_sign_bytes(self):
+        to_sign = b'123'
+        signature = self.credentials.sign_bytes(to_sign)
+        crypt.verify_signature(to_sign, signature, PUBLIC_CERT_BYTES)
+
+    def test_create_scoped(self):
+        credentials = self.credentials.with_scopes(['email', 'profile'])
+        assert credentials._scopes == ['email', 'profile']
+
+    def test__make_authorization_grant_assertion(self):
+        token = self.credentials._make_authorization_grant_assertion()
+        payload = jwt.decode(token, PUBLIC_CERT_BYTES)
+        assert payload['iss'] == self.service_account_email
+        assert payload['aud'] == self.token_uri
+
+    def test__make_authorization_grant_assertion_scoped(self):
+        credentials = self.credentials.with_scopes(['email', 'profile'])
+        token = credentials._make_authorization_grant_assertion()
+        payload = jwt.decode(token, PUBLIC_CERT_BYTES)
+        assert payload['scope'] == 'email profile'
+
+    def test__make_authorization_grant_assertion_subject(self):
+        credentials = self.credentials.with_subject('user@example.com')
+        token = credentials._make_authorization_grant_assertion()
+        payload = jwt.decode(token, PUBLIC_CERT_BYTES)
+        assert payload['sub'] == 'user@example.com'
+
+    @mock.patch('google.oauth2._client.jwt_grant')
+    def test_refresh_success(self, jwt_grant_mock):
+        jwt_grant_mock.return_value = (
+            'token', _helpers.utcnow() + datetime.timedelta(seconds=500), None)
+        request_mock = mock.Mock()
+
+        # Refresh credentials
+        self.credentials.refresh(request_mock)
+
+        # Check jwt grant call.
+        assert jwt_grant_mock.called
+        request, token_uri, assertion = jwt_grant_mock.call_args[0]
```

Comment:
Why use `call_args` here? Try to avoid it at all costs. (I'm not even sure what the 0 index means from `call_args`.)